src/main/scripts/src/app/shared/search-box/search-box.component.ts

```diff
@@ -32,44 +35,55 @@ import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/mergeMap';
 
 import {Suggestions} from '../../core/dto/suggestions';
-import {SearchService} from '../../core/rest/search.service';
 import {WorkspaceService} from '../../core/workspace.service';
 import {QueryItem} from './query-item/query-item';
 import {FulltextQueryItem} from './query-item/fulltext-query-item';
 import {AttributeQueryItem} from './query-item/attribute-query-item';
 import {CollectionQueryItem} from './query-item/collection-query-item';
-import {SuggestionType} from '../../core/dto/suggestion-type';
 import {Collection} from '../../core/dto/collection';
 import {QueryItemsConverter} from './query-item/query-items-converter';
 import {Query} from '../../core/dto/query';
 import {QueryConverter} from '../utils/query-converter';
 import {ViewService} from '../../core/rest/view.service';
 import {KeyCode} from '../key-code';
 import {HtmlModifier} from '../utils/html-modifier';
+import {SearchMockService} from '../../core/rest/search-mock.service';
+import {QueryItemType} from './query-item/query-item-type';
+import {ConditionQueryItem} from './query-item/condition-query-item';
 
 @Component({
   selector: 'search-box',
   templateUrl: './search-box.component.html',
   styleUrls: ['./search-box.component.scss'],
   providers: [QueryItemsConverter]
 })
-export class SearchBoxComponent implements OnInit {
+export class SearchBoxComponent implements OnInit, AfterViewInit {
+
+  @ViewChildren('queryItemCondition')
+  public conditions: QueryList<ElementRef>;
+
+  @ViewChild('searchBox')
+  public input: ElementRef;
 
   public queryItems: QueryItem[] = [];
-  public currentQueryItem: QueryItem = new FulltextQueryItem('');
 
   private searchTerms = new Subject<string>();
-  public suggestions: Observable<QueryItem[]>;
+  private suggestionItems: QueryItem[] = [];
 
-  public text = '';
-  private type = SuggestionType.All;
+  public text: string = '';
+  public shouldShowSuggestions: boolean = false;
+  private shouldFocusCondition: boolean = false;
+  private selectedQueryItem: number = -1;
+  private selectedSuggestion: number = -1;
+  private textCopy: string = '';
 
   constructor(private activatedRoute: ActivatedRoute,
               private queryItemsConverter: QueryItemsConverter,
               private router: Router,
-              private searchService: SearchService,
+              private searchService: SearchMockService,
```

Comment:
I know we agreed on using mock service. But it is not possible to get any meaningful search results at the moment.

Reply:
So how should I do that?

Reply:
Isn't it possible to return real results (collection suggestions) together with mocked ones (attribute suggestions) in `SearchMockService`? If it is too much work then let it be but our application is based mostly on the navigation using the search results and it will not be possible after your changes.

Reply:
I have just realized that attribute suggestions are in fact implemented on the backend so you can make use of it.

Reply:
OK. I will do that.